src/components/OptionsDialog/index.js

```diff
@@ -23,13 +23,30 @@ class OptionsDialog extends React.Component {
     }
     this.onSSHKeyChange = this.onSSHKeyChange.bind(this)
     this.onSaveClick = this.onSaveClick.bind(this)
+    this.setEventHandlers = this.setEventHandlers.bind(this)
   }
 
   componentDidMount () {
+    this.setEventHandlers(true)
+  }
+
+  componentDidUpdate () {
+    this.setEventHandlers()
+  }
+
+  setEventHandlers (mount) {
     const dom = ReactDOM.findDOMNode(this)
-    $(dom).on('shown.bs.modal', function (e) {
+    if (mount) {
+      $(dom).on('shown.bs.modal', function (e) {
+        if (this.props.userId) {
+          this.props.getSSH()
+        }
+      }.bind(this))
+    }
+    $(dom).on('hidden.bs.modal', function (e) {
       if (this.props.userId) {
-        this.props.onOpen()
+        this.props.getSSH()
```

Comment:
I believe this request is redundant. We can save the ssh key into two separate properties of `this.state`, say `sshKey` and `originalSshKey`, in `componentWillReceiveProps` method and later restore original ssh key in `hidden.bs.modal` callback.

```
this.setState({ sshKey: this.state.originalSshKey})
```

Such caching of original ssh key would cause flickering when dialog opened in negligible number of cases.

Reply:
There's no need to create one more state, because props already has original key. Thus I think we need to leave this solution as it is.

Reply:
And it do, on line 49.

Reply:
Ye. So we are ok with unlikely flickering, line 48 can be removed.

Reply:
But no, if someone change SSH key while modal window is hidden, then flickering still be.